docs/spec/v1beta1/provider.md

```diff
@@ -222,3 +223,105 @@ SECRET_NAME=tls-certs
 kubectl create secret generic $SECRET_NAME \
   --from-file=caFile=ca.crt
 ```
+
+### Azure Event Hub
+
+The Azure Event Hub supports two authentication methods, [JWT](https://docs.microsoft.com/en-us/azure/event-hubs/authenticate-application) and [SAS](https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-shared-access-signature) based.
+
+#### JWT based auth
+
+In JWT we use 3 input values.
+
+Channel, token and address. We perform the following translation to match we the data we need to communicate with Azure Event Hub.
+
+* channel = Azure Event Hub namespace
+* address = Azure Event Hub name
+* token   = JWT
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta1
+kind: Provider
+metadata:
+  name: azureeventhub
+spec:
+  type: azureeventhub
+  channel: fluxv2
+  secretRef:
+    name: webhook-url
+```
+
+```yaml
+apiVersion: v1
+data:
+  address: Zmx1eHYy
+  token: QS12YWxpZC1KV1QtdG9rZW4=
+kind: Secret
+metadata:
+  name: webhook-url
+  namespace: default
+type: Opaque
+```
+
+Notification controller doesn't take any responsibility for the JWT token to be updated.
+You need to use a secondary tool to make sure that the token in the secret is renewed.
+
+If you want to make a easy test assuming that you have setup a Azure Enterprise application and you called it event-hub you can follow most of the bellow commands.
+You will need to provide the client_secret that you got when generating the Azure Enterprise Application.
+
+```shell
+export AZURE_CLIENT=$(az ad app list --filter "startswith(displayName,'event-hub')" --query '[].appId' |jq -r '.[0]')
+export AZURE_SECRET='secret-client-secret-generated-at-creation'
+export AZURE_TENANT=$(az account show -o tsv --query tenantId)
+
+curl -X GET --data 'grant_type=client_credentials' --data "client_id=$AZURE_CLIENT" --data "client_secret=$AZURE_SECRET" --data 'resource=https://eventhubs.azure.net' -H 'Content-Type: application/x-www-form-urlencoded' https://login.microsoftonline.com/$AZURE_TENANT/oauth2/token |jq .access_token
```

Comment:
https://docs.microsoft.com/en-us/cli/azure/account?view=azure-cli-latest#az_account_get_access_token

Reply:
This will use your account instead of the application account. Risk of giving to much access, I would recomend to use this way.